csharp/src/Microsoft.ML.OnnxRuntime/OrtFloat16.shared.cs

```diff
@@ -248,17 +249,17 @@ internal static float CreateSingle(bool sign, byte exponent, uint significand)
         /// <summary>
         /// Float16 Positive Infinity value
         /// </summary>
-        public static Float16 PositiveInfinity => new Float16(PositiveInfinityBits);      //  1.0 / 0.0;
+        public static Float16 PositiveInfinity => new Float16(PositiveInfinityBits);
 
         /// <summary>
         /// Float16 Negative Infinity value
         /// </summary>
-        public static Float16 NegativeInfinity => new Float16(NegativeInfinityBits);      // -1.0 / 0.0
+        public static Float16 NegativeInfinity => new Float16(NegativeInfinityBits);
 
         /// <summary>
         /// Float16 NaN
         /// </summary>
-        public static Float16 NaN => new Float16(NegativeQNaNBits);                       //  0.0 / 0.0
+        public static Float16 NaN => new Float16(PositiveQNaNBits);                       //  quiet NaN
```

Comment:
The issue with this change, is that now it is going to diverge with the corresponding `System.Half` class introduced in the later version of .NET framework.
It is a good thing to be consistent in the product implementation across the interfaces, but C# coders may disagree.

https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Half.cs#L95

Said that, IsNaN() would still work properly.

The alternative proposal to fix C++ code to test for Nans in a more comprehesive manner.

Reply:
C++ on Linux also has a builtin float16 type: https://gcc.gnu.org/onlinedocs/gcc/Half-Precision.html

Reply:
I thinking about this scenario.

1) when results returned with NaN from C++ C# would be able to test it with IsNaN() no issues.

2) if the user feeds data from C# to C++ and it has NaN defined as NegativeNaN, would C++ be able to detect that properly. If so, no issues.